zzk/connection.go

```diff
@@ -38,6 +40,34 @@ var (
 // GetConnection describes a generic function for acquiring a connection object
 type GetConnection func(string) (client.Connection, error)
 
+type delay struct {
+	backoff time.Duration
+}
+
+func (d *delay) GetDelay() time.Duration {
+	defer func() {
```

Comment:
While creating _delay_ object user needs to give _baseDelay_ as argument. I want to return the same value for the first time when _GetDelay_ is being called, thus this `defer` is here